tools/python/run_CIs_for_branch.py

```diff
@@ -44,12 +47,30 @@ def _parse_args():
         """,
     )
 
-    parser.add_argument("-i", "--include", type=str, help="Include CIs that match this string. Case insensitive.")
-    parser.add_argument("-e", "--exclude", type=str, help="Exclude CIs that match this string. Case insensitive.")
+    current_branch = None
+    get_branch_result = subprocess.run(["git", "branch", "--show-current"], capture_output=True, text=True, check=False)
+    if get_branch_result.returncode == 0:
+        current_branch = get_branch_result.stdout.strip()
+
+    parser.add_argument(
+        "-i", "--include", action="append", type=str, help="Include CIs that match this string. Case insensitive."
+    )
+    parser.add_argument(
+        "-e", "--exclude", action="append", type=str, help="Exclude CIs that match this string. Case insensitive."
+    )
     parser.add_argument("--dry-run", action="store_true", help="Print selected CIs but do not run them.")
-    parser.add_argument("branch", type=str, help="Specify the branch to run.")
+    parser.add_argument(
+        "branch",
+        type=str,
+        nargs="?",
+        default=current_branch,
+        help="Specify the branch to run. Default is current branch is available.",
+    )
 
     args = parser.parse_args()
+    if not args.branch:
+        raise ValueError("Branch was unable to be inferred and must be specified")
```

Comment:
nit: maybe attempt to get the current branch only when `branch` is not specified instead of unconditionally?

Reply:
If I do that it can't be displayed in the help as the default value. I swallow any errors getting the branch and the `git branch` command is fast so I'd slightly prefer the current setup..

Reply:
oh, displaying it in the help would be nice. it doesn't get shown currently. maybe put it directly in the `help` argument?

```
> python .\tools\python\run_CIs_for_branch.py --help
2024-03-07 17:49:15,952 tools_python_utils [INFO] - flatbuffers module is not installed. parse_config will not be available
usage: run_CIs_for_branch.py [-h] [-i INCLUDE] [-e EXCLUDE] [--dry-run] [branch]

Run the CIs used to validate PRs for the specified branch.

        If not specified, the branch will be inferred (if possible) by running `git branch --show-current`.

        If specified, the `--include` filter is applied first, followed by any `--exclude` filter.
        `--include` and `--exclude` can be specified multiple times to accumulate values to include/exclude.

        Requires the Azure CLI with DevOps extension to be installed.
          Azure CLI: https://learn.microsoft.com/en-us/cli/azure/install-azure-cli
          DevOps extension: https://github.com/Azure/azure-devops-cli-extension

        Configuration:
          Login:`az login`
          Configure ORT repo as default:
            `az devops configure --defaults organization=https://dev.azure.com/onnxruntime project=onnxruntime`

        Example usage:
          List all CIs
            `python run_CIs_for_branch.py --dry-run my/BranchName`
          Run all CIs
            `python run_CIs_for_branch.py my/BranchName`
          Run only Linux CIs
            `python run_CIs_for_branch.py --include linux my/BranchName`
          Exclude training CIs
            `python run_CIs_for_branch.py --exclude training my/BranchName`
          Run non-training Linux CIs
            `python run_CIs_for_branch.py --include linux --exclude training my/BranchName`


positional arguments:
  branch                Specify the branch to run. Default is current branch if available.

optional arguments:
  -h, --help            show this help message and exit
  -i INCLUDE, --include INCLUDE
                        Include CIs that match this string. Case insensitive.
  -e EXCLUDE, --exclude EXCLUDE
                        Exclude CIs that match this string. Case insensitive.
  --dry-run             Print selected CIs but do not run them.
```

Reply:
Merged formatters to do that. 

![image](https://github.com/microsoft/onnxruntime/assets/979079/e4ce84d3-49d4-4012-a5b6-b48397a29638)